src/librustdoc/clean/mod.rs

```diff
@@ -2373,21 +2374,22 @@ fn clean_maybe_renamed_item<'tcx>(
             _ => unreachable!("not yet converted"),
         };
 
-        let mut extra_attrs = Vec::new();
+        let mut import_attrs = Vec::new();
```

Comment:
I'd nitpick to rename it into `import_attrs` but feel free to ignore it.

Reply:
If there are multiple imports there, then there should be multiple `import_id`s and their parents, so there's still a lot to fix here besides the naming.

Reply:
Just to be sure we're talking about this same thing:

```rust
mod foo {
    struct A;

    pub use self::A as B;
    pub use self::B as C;
}
pub use foo::C as D;
```

Intermediate imports are `B` and `C` and they are the ones we're gathering attributes from.

Reply:
Yes, and their attributes should not be all merged together, but should instead be broken into groups with a separate group with a separate parent id for every `use` item in the reexport chain.

Reply:
Do you have cases where it would be problematic by any chance?

Reply:
Same as the test in this PR, but with a doc link on an intermediate import.

Reply:
Ok, I'm gonna send a fix for that too in the next days. Thanks for fixing this bug already.

Reply:
I'm not sure how many places in rustdoc merge attributes from different items, but all of them should ideally apply the same fix - a separate parent per item.